packages/core/src/lib/select/searchable-select.svelte

```diff
@@ -88,26 +96,57 @@ const FOCUS_ITEM = 'focus-item';
 
 type MenuState = typeof CLOSED | typeof FOCUS_SEARCH | typeof FOCUS_ITEM;
 
+const detailedOptions = writable(optionsToDetailedOptions(options));
+$: detailedOptions.set(optionsToDetailedOptions(options));
+
+const getDetailedOption = (val: string) =>
+  $detailedOptions.find((opt) => opt.value === val);
+
 const optionElements: Record<string, HTMLElement> = {};
+
 let inputElement: HTMLInputElement | undefined;
 let autoSelectIndex = -1;
 let menuState: MenuState | undefined;
+// previousValue is used to ensure we don't double-call onChange
 let previousValue: string | undefined = undefined;
+// initialize the search value to the option that matches the passed value field (or the value field itself as a fallback)
+let searchValue = optionDisplayValue(getDetailedOption(value) ?? { value });
```

Comment:
This is not reactive to value changing. If you think there is a use case where it should be, I can make it reactive.

Reply:
fd0802e